Beginner-Installation-Guide.md

```diff
@@ -0,0 +1,287 @@
+    wget \
+    libusb-devel \
+    cmake \
+    pv \
+    bsdiff \
+    diffutils \ 
+    texinfo
+```
+
+Install `qemu` and `qt5-devel` (for emulation and analysis with UEFITool respectively):
+
+```
+dnf install -y \
+    qemu qt5-devel \
+```
+
+Clone the Heads repo:
+
+```
+git clone https://github.com/osresearch/heads.git
+```
+
+Move into the cloned repository:
+
+```
+cd heads
+```
+
+Builds of Heads are reproducible, which means that Heads will build in the exact same way on different computers. Because of this, as a user, you can guarantee that Heads has built correctly and has not been tampered with.
+
+However, this also means that the first time you build Heads it must first build the compilers that it will use to build itself. If that seems complicated, don’t worry. The result is that the first build of Heads will take about an hour to complete. After the first build, building Heads will take less than a minute.
+
+For the Thinkpad x230 there are two boards in `./boards`, x230-flash and x230. x230-flash is externally flashable and contains a smaller package that will let us boot to the Heads recovery shell. x230 is only internally flashable and contains all of Heads. Since we will end up using both x230-flash and x230, it makes the most sense to build both now.
+
+Make both boards:
+
+```
+make BOARD=x230-flash
+make BOARD=x230
+```
+
+Make Heads for another board (`XXX` should be the name of your board in ./boards):
+
+```
+make BOARD=XXX
+```
+
+The resulting rom file can be found in `./build/XXX/XXX.rom` (`XXX` should be the name of your board in `./boards`).
+
+
+## External Flashing:
+
+First remove the battery or cable powering your device. The Thinkpad x230 has two SPI flash chips that hold the BIOS, ME, etc. and are located under the palm rest. To access these chips, first remove the indicated screws on the back of the laptop.
+
+Removing these screws will allow you to remove the keyboard and palm rest. 
+
+The keyboard is connected to the motherboard by a ribbon cable which easily detaches from the motherboard. (The keyboard only needs to be removed so that the palm rest can be removed. After removing the palm rest, you can put the keyboard back.) 
+
+The palm rest is also connected to the motherboard, but there is a little latch holding its ribbon cable. After undoing that latch, the palm rest should be fairly easy to remove.
+
+Pull up the black plastic on the bottom right of the palm rest to reveal the two SPI flash chips.
+
+The top chip is 4MB and contains the BIOS and reset vector. The bottom chip is 8MB and has the [Intel Management Engine (ME)](https://www.flashrom.org/ME) firmware, plus the flash descriptor.
+
+Try to read the name on the top SPI flash chip. Then, connect the clip and ch341a programmer to the top SPI flash chip. Use flashrom to check the chip that you are connected to:
+
+```
+sudo flashrom -p ch341a_spi
+```
+
+Find the chip and read from it twice (For me the SPI flash chip is `YYY`):
+
+```
```

Comment:
I would recommend doing -v instead to verify rom image against of rereading and diffing